matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java

```diff
@@ -684,7 +684,7 @@ public void joinWithThirdPartySigned(final String alias, final String thirdParty
             join(alias, callback);
         } else {
             String url = thirdPartySignedUrl + "&mxid=" + mMyUserId;
-            UrlPostTask task = new UrlPostTask();
+            UrlPostTask task = new UrlPostTask(mDataHandler.getHomeServerConnectionConfig());
```

Comment:
Please just add a parameter `MXSession session` to the method, as it is done to some other methods of this class, to retrieve the homeServerConnectionConfig from it.
It's ok for the API break, I will do the PR on Riot-Android to handle it.